### PR TITLE
fix: Actually use node_pool_vm_count

### PR DESCRIPTION
**Pull request by yogeshg**

### CODE REVIEW

**Review by jsbroks — APPROVED** [review]

File: modules/app_aks/main.tf [hunk 1]
```diff
@@ -18,8 +18,8 @@ resource "azurerm_kubernetes_cluster" "default" {
 
   default_node_pool {
     name                = "default"
-    node_count          = 2
-    vm_size             = "Standard_D4s_v3"
+    node_count          = var.node_pool_vm_count
+    vm_size             = var.node_pool_vm_size
```
Comment (yogeshg) [inline modules/app_aks/main.tf:22]:
```suggestion
    vm_size             = var.node_pool_vm_size 
```

File: modules/app_aks/main.tf [hunk 2]
```diff
@@ -18,8 +18,8 @@ resource "azurerm_kubernetes_cluster" "default" {
...
```
Comment (yogeshg) [inline modules/app_aks/main.tf:22]:
```suggestion
    vm_size             = var.node_pool_vm_size
```

File: modules/app_aks/main.tf [hunk 3]
```diff
@@ -17,13 +17,14 @@ resource "azurerm_kubernetes_cluster" "default" {
...
-    enable_auto_scaling = false
-    zones               = ["1", "2"]
+    name                        = "default"
+    node_count                  = var.node_pool_vm_count
+    vm_size                     = var.node_pool_vm_size
+    vnet_subnet_id              = var.cluster_subnet_id
+    type                        = "VirtualMachineScaleSets"
+    enable_auto_scaling         = false
+    zones                       = ["1", "2"]
+    temporary_name_for_rotation = "foo"
```
Comment (yogeshg) [inline modules/app_aks/main.tf:27]:
```suggestion
    temporary_name_for_rotation = "temporary_nodes_for_rotation"
```

File: modules/app_aks/main.tf [hunk 4]
```diff
@@ -17,13 +17,14 @@ resource "azurerm_kubernetes_cluster" "default" {
...
-    enable_auto_scaling = false
-    zones               = ["1", "2"]
+    name                        = "default"
+    node_count                  = var.node_pool_vm_count
+    vm_size                     = var.node_pool_vm_size
+    vnet_subnet_id              = var.cluster_subnet_id
+    type                        = "VirtualMachineScaleSets"
+    enable_auto_scaling         = false
+    zones                       = ["1", "2"]
+    temporary_name_for_rotation = "temporary_nodes_for_rotation"
```
Comment (yogeshg) [inline modules/app_aks/main.tf:27]:
```suggestion
    temporary_name_for_rotation = "rotating"
```

File: modules/app_aks/main.tf [hunk 5]
```diff
@@ -17,13 +17,14 @@ resource "azurerm_kubernetes_cluster" "default" {
...
-    enable_auto_scaling = false
-    zones               = ["1", "2"]
+    name                        = "default"
+    node_count                  = var.node_pool_vm_count
+    vm_size                     = var.node_pool_vm_size
+    vnet_subnet_id              = var.cluster_subnet_id
+    type                        = "VirtualMachineScaleSets"
+    enable_auto_scaling         = false
+    zones                       = ["1", "2"]
+    temporary_name_for_rotation = "rotating"
```
Comment (yogeshg) [inline modules/app_aks/main.tf:27]: this param is needed when changing vm_size.

**Comment by jsbroks** [comment]
This PR is included in version 2.1.1 :tada:

**Pull request merged**